### PR TITLE
Send connectivity test result to specified report collector

### PR DESCRIPTION
**Pull request by amircybersec**
I made a new branch for this since I messed up the previous branch when doing unnecessary rebase and addressing merge conflicts :( 

I moved all the changes to this branch and also implement failure and success sample rate flags to send reports with a given probability. I also added an optional port flag. There should not be any merge conflict with merge since I fetched upstream before making this branch.

I am going to close the other pull request.

### CODE REVIEW

File: x/examples/outline-connectivity/main.go [hunk 1]
```diff
@@ -98,8 +132,26 @@ func main() {
 	domainFlag := flag.String("domain", "example.com.", "Domain name to resolve in the test")
 	resolverFlag := flag.String("resolver", "8.8.8.8,2001:4860:4860::8888", "Comma-separated list of addresses of DNS resolver to use for the test")
 	protoFlag := flag.String("proto", "tcp,udp", "Comma-separated list of the protocols to test. Must be \"tcp\", \"udp\", or a combination of them")
+	reportToFlag := flag.String("report-to", "", "URL to send JSON error reports to")
+	portFlag := flag.String("port", "53", "Resolver port to use for the test")
```
Comment (fortuna) [inline x/examples/outline-connectivity/main.go:136]: Let's remove this flag for now, or add the port to the resolverFlag list instead.

**Comment by amircybersec** [comment]
@fortuna I was thinking to drop this PR and refactor the code using the reporter package I wrote here:

https://github.com/Jigsaw-Code/outline-sdk/pull/116

Since I am using the same logic in GUI Apps, this would help clean things up.

File: x/examples/outline-connectivity/main.go (hunk 1 above)
Comment (amircybersec) [reply x/examples/outline-connectivity/main.go]: addressed in 69f4e9278e8c54cb69e750fca0e7b3a7d5859fbb

File: x/examples/outline-connectivity/main.go [hunk 2]
```diff
@@ -159,6 +210,28 @@ func main() {
...
 				log.Fatalf("Failed to output JSON: %v", err)
 			}
+			// Send error report to collector if specified
+			if *reportToFlag != "" {
+				var samplingRate float64
+				if !success {
+					samplingRate = *reportFailureFlag
+				} else {
+					samplingRate = *reportSuccessFlag
+				}
```
Comment (fortuna) [inline x/examples/outline-connectivity/main.go:220]: This is easier to reason about, so we don't need to think about a double-negative in the else case.
```suggestion
				if success {
					samplingRate = *reportSuccessFlag
				} else {
					samplingRate = *reportFailureFlag
				}
```

File: x/examples/outline-connectivity/main.go [hunk 3]
```diff
@@ -159,6 +210,28 @@ func main() {
...
+				// Generate a random number between 0 and 1
+				random := rand.Float64()
+				if random < samplingRate {
+					// Run your function here
```
Comment (fortuna) [inline x/examples/outline-connectivity/main.go:224]: Remove?
```suggestion
```

File: x/examples/outline-connectivity/main.go [hunk 4]
```diff
@@ -159,6 +210,28 @@ func main() {
...
+					err = sendReport(record, *reportToFlag)
+					if err != nil {
+						log.Fatalf("HTTP request failed: %v", err)
```
Comment (fortuna) [inline x/examples/outline-connectivity/main.go:227]: This would be clearer:
```suggestion
						log.Fatalf("Report failed: %v", err)
```

**Review by fortuna — COMMENTED** [review]
Looks good. A few things below.

**Review by fortuna — APPROVED** [review]

**Comment by fortuna** [comment]
Let's merge this, and iterate on the reporter library.

**Pull request merged**